demos/http/http_demo_s3_generate_presigned_url/http_demo_s3_generate_presigned_url.c

```diff
@@ -0,0 +1,643 @@
+    ( void ) memset( &requestHeaders, 0, sizeof( requestHeaders ) );
+    ( void ) memset( &requestInfo, 0, sizeof( requestInfo ) );
+    ( void ) memset( &response, 0, sizeof( response ) );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = serverHost;
+    requestInfo.hostLen = serverHostLength;
+    requestInfo.pMethod = HTTP_METHOD_GET;
+    requestInfo.methodLen = HTTP_METHOD_GET_LENGTH;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. This is done in
+     * order to download the file in parts. */
+    requestInfo.reqFlags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    /* Initialize the response object. The same buffer used for storing request
+     * headers is reused here. */
+    response.pBuffer = userBuffer;
+    response.bufferLen = USER_BUFFER_LENGTH;
+
+    /* Generate and print the pre-signed URL. */
+    returnStatus = printS3ObjectFilePresignedURL( pTransportInterface,
+                                                  serverHost,
+                                                  serverHostLength,
+                                                  pPath );
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool printS3ObjectFilePresignedURL( const TransportInterface_t * pTransportInterface,
+                                           const char * pHost,
+                                           size_t hostLen,
+                                           const char * pPath )
+{
+    bool returnStatus = true;
+    HTTPStatus_t httpStatus = HTTPSuccess;
+    HTTPRequestHeaders_t requestHeaders;
+    HTTPRequestInfo_t requestInfo;
+    HTTPResponse_t response;
+    uint8_t userBuffer[ USER_BUFFER_LENGTH ];
+
+    SigV4Status_t sigv4Status = SigV4Success;
+    SigV4HttpParameters_t sigv4HttpParams;
+
+    char * pHeaders = NULL;
+    size_t headersLen = 0;
+
+    /* Store Signature used in AWS HTTP requests generated using SigV4 library. */
+    char * signature = NULL;
+    size_t signatureLen = 0;
+
+    assert( pHost != NULL );
+    assert( pPath != NULL );
+
+    /* Initialize all HTTP Client library API structs to 0. */
+    ( void ) memset( &requestHeaders, 0, sizeof( requestHeaders ) );
+    ( void ) memset( &requestInfo, 0, sizeof( requestInfo ) );
+    ( void ) memset( &response, 0, sizeof( response ) );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = hostLen;
+    requestInfo.pMethod = HTTP_METHOD_GET;
+    requestInfo.methodLen = sizeof( HTTP_METHOD_GET ) - 1;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
```

Comment:
Why set these values to a local variable here when `generateS3ObjectFilePresignedURL` sets them to a global?